src/main/java/bio/terra/service/common/JournalRecordDeleteEntryStep.java

```diff
@@ -31,20 +32,23 @@ public JournalRecordDeleteEntryStep(
     this.resourceKey = resourceKey;
     this.resourceType = resourceType;
     this.note = note;
-    this.journalEntryKey = null;
   }
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
-    this.journalEntryKey =
+    FlightMap workingMap = context.getWorkingMap();
+    UUID journalEntryKey =
         journalService.recordDelete(
             userReq, resourceKey, resourceType, note, getFlightInformationOfInterest(context));
+    workingMap.put(JobMapKeys.JOURNAL_ENTRY_KEY.getKeyName(), journalEntryKey);
     return StepResult.getStepResultSuccess();
   }
 
   @Override
   public StepResult undoStep(FlightContext context) throws InterruptedException {
-    Optional.ofNullable(this.journalEntryKey).ifPresent(this.journalService::removeJournalEntry);
+    FlightMap workingMap = context.getWorkingMap();
+    UUID journalEntryKey = workingMap.get(JobMapKeys.JOURNAL_ENTRY_KEY.getKeyName(), UUID.class);
+    Optional.ofNullable(journalEntryKey).ifPresent(journalService::removeJournalEntry);
```

Comment:
At this point the `journalEntryKey` shouldn't be null right? I wasn't sure if it was safe to remove this.

Reply:
I think it makes sense to leave this check in based on what you said! Also if the journal entry key isn't in the flight map, this will return a null UUID.

Reply:
perfect, thanks for checking!

Reply:
FWIW this is doing what stairway docs say you should _not_ do. The docs warn against trying to use the context map to communicate between `doStep()` and `undoStep()` and specifically talk about the case of creating a database entry. In that case, the recommended approach is to have a separate step that creates the UUID and stores it in the context map. See [Working Map Best Practices](https://github.com/DataBiosphere/stairway/blob/develop/FLIGHT_DEVELOPER_GUIDE.md#working-map-best-practices-) and [Database Transactions](https://github.com/DataBiosphere/stairway/blob/develop/FLIGHT_DEVELOPER_GUIDE.md#database-transactions-)

Reply:
Yeah, though I would argue that this ends up having pretty much the same effect

Reply:
> Yeah, though I would argue that this ends up having pretty much the same effect

I'm not so sure. One thing that stairway docs mention that I'm not so sure we're good about is making the steps (both do and undo) idempotent, since they can fail at any point up to when the step is registered as complete in the stairway database. If the code to create a journal entry does a database insert and has the db generate the ID, then you could end up with multiple records in the db. If instead it was using an ID generated by an upstream task then this problem wouldn't occur. (If the db create relies on other unique field constraints to prevent duplicate records, you "fix" this problem but then have the problem where you have to ignore the field constraint error in case the step has already been run in the current flight.)

Reply:
Ah yeah, I see your point. Sigh, it kills be that we need to have a second step for this kind of stuff.  This feels like a stairway flaw for re-usability.  Like, we just have this simple step and I hate the idea that you'd need a separate step in order for this one generic step to work.  It should be able to all live within one step